src/cr/cube/matrix.py

```diff
@@ -55,22 +86,83 @@ def rows(self):
 
     @lazyproperty
     def table_base(self):
+        """np.int64 scalar or a 1D or 2D ndarray of np.int64 representing table base.
+
+        A multiple-response (MR) dimension produces an array of table-base values
+        because each element (subvariable) of the dimension represents a logically
+        distinct question which may not have been asked of all respondents. When both
+        dimensions are MR, the return value is a 2D ndarray and there is a distinct
+        table-base value for each "cell" of the matrix. A CAT_X_CAT matrix produces
+        a scalar value for this property.
+        """
         return self.table_base_unpruned[
             np.ix_(self._visible_rows_mask, self._visible_cols_mask)
         ]
 
     @lazyproperty
     def table_base_unpruned(self):
```

Comment:
We should consider moving ‘prune’ to be a computed-hide transform so vectors can only be inserted/removed via the transform route

Reply:
Yeah, that's sounds like a good idea. We could probably add it to the dimension transforms schema, maybe `"prune": true`. I know end-users think of it as show-empty with opposite boolean semantics, but once we're inside the technical realm, "prune" means do something and "don't prune" means "don't do something" so it's less of a brain-twister to interpret the semantics.

An possible advantage of dimension-level specification would be you could say to prune rows but not columns. It would default to `true` so you wouldn't often need to say it in two places. Also an advantage is we wouldn't need to create a new "table" level transforms container to put this in; so far we haven't had a need for transforms at that level.